score-server/src/test/java/bio/overture/score/server/security/JWTSecurityTest.java

```diff
@@ -0,0 +1,265 @@
+package bio.overture.score.server.security;
+
+import bio.overture.score.core.model.ObjectSpecification;
+import bio.overture.score.server.config.SecurityConfig;
+import bio.overture.score.server.metadata.MetadataEntity;
+import bio.overture.score.server.metadata.MetadataService;
+import bio.overture.score.server.repository.DownloadService;
+import bio.overture.score.server.repository.UploadService;
+import bio.overture.score.server.utils.JWTGenerator;
+import bio.overture.score.server.utils.JwtContext;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import java.security.KeyPair;
+import java.util.List;
+
+import static bio.overture.score.server.security.JWTSecurityTest.RequestType.DOWNLOAD;
+import static bio.overture.score.server.security.JWTSecurityTest.RequestType.UPLOAD;
+import static bio.overture.score.server.security.JWTSecurityTest.ScopeOptions.*;
+import static bio.overture.score.server.utils.JwtContext.buildJwtContext;
+import static java.util.Objects.isNull;
+import static org.apache.commons.lang.StringUtils.isBlank;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.when;
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
+
+@Slf4j
+@SpringBootTest
+@ContextConfiguration
+@RunWith(SpringJUnit4ClassRunner.class)
+@ActiveProfiles({"test", "secure", "jwt", "default", "dev"})
+public class JWTSecurityTest {
```

Comment:
In addition to the sad path tests, we should have happy path tests. I have attached a spreadsheet with tests following the naming convention in the previous comments

[PR-269-Jaser-ScoreJWTTests.xlsx](https://github.com/overture-stack/score/files/5087596/PR-269-Jaser-ScoreJWTTests.xlsx)

Reply:
Wow! The spread sheet is very helpful :+1:. I do have one question though, should expired tokens return `403-FORBIDDEN`? Spring security defaultTokenService returns `401-UNAUTHORIZED` which is what the oauth2 spec defines https://tools.ietf.org/html/rfc6750#section-3.1?

Reply:
Yes i think this makes sense. Expired should be 401 not 403. Good find.